### ISSUE TITLE
APSFOUND-30: Automatically compress http responses

### ISSUE THREAD

**Issue by pnicolucci** [OpenLiberty/open-liberty#7502]
Design Document: https://ibm.box.com/s/lkfrv86n9xoxbpkx1dnckd23suzs9uau
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)
 
Instructions:
 - Do the actions below and mark them complete in the checklist when they are done.
 - Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **Before Development Starts or 8 weeks before Onboarding** 
- [x] POC Design / WAD Review Scheduled (David Chang) or N/A.
- [x] POC Design / WAD Reviewed (Feature Owner) or N/A.
- [x] Complete any follow-ons from the POC Review.
- [x] Design / WAD Approval (Alasdair Nottingham) or N/A.
- [x] No Design / No WAD Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or N/A.
- [x] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [x] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [x] Create a child task of the epic entitled "FAT Approval Test Summary".  Add and fill in the template as described here: https://github.ibm.com/was-liberty/WS-CD-Open/wiki/Feature-Review-(Feature-Test-Summary-Process)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **3 weeks before Onboarding** 
- [x] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE ** 3 weeks before Onboarding** 
- [x] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **2 weeks before Onboarding** 
- [x] Implementation complete. (Epic owner / Feature owner) 
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [x] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## APPROVALS with TARGET COMPLETION DATE **2 to 1 week before Onboarding** 
Prereq: You must have the Design Approved or No Design Approved label on the GitHub Epic.
- [x] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [x] **FAT Liberty SOE** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [x] **Globalization** (Marika Joannidis - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.
- [x] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [x] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [x] **Serviceability** - (Don Bourne). Serviceability has been addressed. 
- [x] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [x] **SVT** - (Greg Ecock - Cloud, Brian Hanczaryk- APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.
- [x] **Demo** - (Liberty only - Tom Evans or Chuck Bridgham). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## TARGET COMPLETION DATE **1 week before Onboarding** 
- [x] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [x] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [x] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## NOT REQUIRED FOR A FEATURE
- [x] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A. 
- [x] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
 
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Related Deliverables TARGET COMPLETION DATE **General Availability** 
- [x] Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by atosak** [comment]
https://github.com/OpenLiberty/open-liberty/issues/8495

**Comment by pnicolucci** [comment]
FAT Approval Test Summary - APSFOUND-30 : #9020

**Comment by atosak** [comment]
Targeting to Have this as Beta in 20001.

**Comment by pnicolucci** [comment]
Feature Review Comments:

- [ ] Customer Overview: More value / user oriented (Eric)

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same STE Archive BOX folder.  Thanks!

**Comment by pnicolucci** [comment]
Stand alone blog entry that needs to be completed: https://github.com/OpenLiberty/blogs/issues/291

**Comment by chirp1** [comment]
Paul indicated that doc related updates are in autogen and the blog post. No updates are needed from ID. Approving.

**Comment by pnicolucci** [comment]
Talked with Yee-Kang and there is no guild requirement here.

**Comment by pnicolucci** [comment]
Talked with Erin Harris for WDT and we list the values in the description for the metatype so no action necessary:

> 
http.compression.serverPreferredAlgorithm.desc=The configured compression algorithm is used to compress the body of responses when it is specified with a non-zero quality value in the request's Accept-Encoding header. The valid compression algorithms include: deflate, gzip, x-gzip, zlib, and identity.

**Comment by chirp1** [comment]
Paul provided ID info in https://github.ibm.com/was-liberty/twas-docs/issues/2958. Approving
